wenet/efficient_conformer/encoder.py

```diff
@@ -355,22 +364,28 @@ def forward_chunk(
         # using downsampling factor to recover offset
         offset *= self.calculate_downsampling_factor(self.num_blocks + 1)
 
-        # tmp_masks is just for interface compatibility
-        tmp_masks = torch.ones(1,
-                               xs.size(1),
-                               device=xs.device,
-                               dtype=torch.bool)
-        tmp_masks = tmp_masks.unsqueeze(1)  # (1, 1, xs-time)
+        chunk_masks = torch.ones(1,
+                                 xs.size(1),
+                                 device=xs.device,
+                                 dtype=torch.bool)
+        chunk_masks = chunk_masks.unsqueeze(1)  # (1, 1, xs-time)
+
+        real_len = 0
+        if self.global_chunk_size > 0:
+            # for ONNX decode simulation， padding xs to chunk_size
+            real_len = xs.size(1)
+            pad_len = self.chunk_feature_map - real_len
+            xs = F.pad(xs, (0, 0, 0, pad_len), value=0.0)
+            chunk_masks = F.pad(chunk_masks, (0, pad_len), value=0.0)
+
```

Comment:
Out of curiosity, will padding only be applied to the last chunk, given that previous chunks always have a valid chunk size?

Reply:
Yes, padding will only be applied to the last chunk. Also, this part is currently only valid if you manually specify `use_onnx=True`, to simulate the CER after exporting ONNX.